docs/source/developers/python.rst

```diff
@@ -678,6 +642,45 @@ PyArrow are:
      - Number of processes used to compile PyArrow’s C++/Cython components
      - ``''``
 
+The components being disabled or enabled when building PyArrrow is by default
+based on how Arrow C++ is build (i.e. it follows the ``ARROW_$COMPONENT`` flags).
+However, the ``PYARROW_WITH_$COMPONENT`` environment variables can still be used
+to override this when building PyArrow (e.g. to disable components, or to enforce
+certain components to be built):
+
+.. list-table::
+   :widths: 30 30
+   :header-rows: 1
+
+   * - Arrow flags/options
+     - Corresponding environment variables for PyArrow
```

Comment:
This list is bound to become outdated. Could we link to (or auto generate from) the relevant CMake file?

Reply:
We don't really have a good place we could link to, because in the CMakeLists file, the env variables are not directly visible (they are constructed and checked in a macro)